planner/core/planbuilder.go

```diff
@@ -227,6 +227,8 @@ func (b *PlanBuilder) buildDo(v *ast.DoStmt) (Plan, error) {
 			RetType:  expr.GetType(),
 		})
 	}
+	//make dual schema if not
+	var _ = dual.Schema()
```

Comment:
how about directly set the schema of the `dual` operator, which eliminates the confusion:
```go
dual.schema = expression.NewSchema()
```

Reply:
Yes, it looks better